sdk/core/System.Net.ClientModel/README.md

```diff
@@ -0,0 +1,68 @@
+# System.Net.ClientModel shared client library for .NET
+
+`System.Net.ClientModel` provides shared primitives, abstractions, and helpers for .NET service client libraries.
+
+`System.Net.ClientModel` allows client libraries built from its components to expose common functionality in a consistent fashion, so that once you learn how to use these APIs in one client library, you'll know how to use them in other client libraries as well.
+
+## Getting started
+
+Typically, you will not need to install `System.Net.ClientModel`.
+it will be installed for you when you install one of the client libraries using it.
+
+### Install the package
+
+Install the client library for .NET with [NuGet](https://www.nuget.org/).
+
+### Prerequisites
+
+None needed for `System.Net.ClientModel`.
+
+### Authenticate the client
+
+The `System.Net.ClientModel` preview package provides a `KeyCredential` type for authentication.
+
+## Key concepts
+
+The main shared concepts of `System.Net.ClientModel` include:
+
+- Configuring service clients (`RequestOptions`).
+- Accessing HTTP response details (`Result`, `Result<T>`).
+- Exceptions for reporting errors from service requests in a consistent fashion (`MessageFailedException`).
+
+## Examples
+
+### Send a message using the MessagePipeline
+
+A rudimentary client implementation is as follows:
+
+```csharp
```

Comment:
For a GA library, we'll need to generate snippets from samples that are part of tests.

@m-nash, please add a samples directory to the public test projects - we should be able to follow patterns for existing client and/or core libraries on this.

We'll want to be clear on specifically which samples we illustrate for customers per mainline library consumer scenarios, and client author scenarios.

Reply:
Originally we agreed to postpone on this until we ship the usages.  https://github.com/Azure/azure-sdk-for-net/issues/38077.  Do you think we should change this decision?